circuit_knitting/cutting/automated_cut_finding.py

```diff
@@ -137,6 +137,8 @@ class OptimizationParameters:
     seed: int | None = OptimizationSettings().seed
     max_gamma: float = OptimizationSettings().max_gamma
     max_backjumps: None | int = OptimizationSettings().max_backjumps
+    gate_lo: bool = OptimizationSettings().gate_lo
```

Comment:
One might want to investigate what happens when one sets both `gate_lo` and `gate_lo` to `False`.  Ideally this will results in an informative error message (right?).